src/editor/editor.cpp

```diff
@@ -265,7 +265,12 @@ Editor::update(float dt_sec, const Controller& controller)
     m_particle_editor_request = false;
     std::unique_ptr<Screen> screen(new ParticleEditor());
     if (m_particle_editor_filename)
-      static_cast<ParticleEditor*>(screen.get())->open("particles/" + *m_particle_editor_filename);
+    {
+      // realpath() is necessary for particle files outside the particles/ folder
+      std::string path = physfsutil::realpath("particles/" + *m_particle_editor_filename);
+
+      static_cast<ParticleEditor*>(screen.get())->open(path);
```

Comment:
You really need to static cast here? You can store the new ParticleEditor in a variable and use that. In my opinion, that would be more readable. And then you can std::make_unique in the push_screen function

Reply:
The static cast was already there. I've copied the code and changed as little as possible so as not to transform the code too much, since I'm not very familiar with the current coding standards and people's preferences. A static cast will be needed either way if `screen`'s type is a pointer to Scene; I can move the static cast on its own line but I can't really avoid it without changing more code. I can't make_unique the screen directly in push_screen because I need to call the `open` function on the scene before pushing it.